src/Interfaces.ts

```diff
@@ -1,72 +1,136 @@
-export interface Claim<T extends ClaimAttributes = ClaimAttributes> {
+import * as Joi from 'joi'
+import JSIG = require('jsonld-signatures')
+
+const jsigs = JSIG()
+
+export enum ClaimType {
+  Identity = 'Identity',
+  Work = 'Work',
+}
+
+export interface Claim {
   readonly '@context'?: any
   readonly id?: string
-
-  readonly issuer?: string
-  readonly issuanceDate?: Date // should replace dateCreated... or created... once we remove protobufs
-  readonly publicKey?: string
-  readonly signature?: string
-  readonly created?: string
+  readonly issuer: string
```

Comment:
Instead of adding these properties to Claim we can extend Claim and create a SignedClaim interface. We can then separate createClaim and signClaim functions.

Properties I am referring to: issuer, issuanceDate and sec:proof. These are all properties that IMO belong to signing a claim and should be added to the claim via signClaim, not by createClaim.

Right now createClaim feels really awkward because its doing more than its supposed to.

I can create a PR tomorrow showing this change in the README and code if it will help

Reply:
The issue is whether those fields belong in the calculated id. I think the issuer of the claIm IS an integral part of the claim itself, and in the long term will not be a data form of the publicKey.